src/Libraries/Nop.Services/Catalog/ProductService.cs

```diff
@@ -240,8 +240,6 @@ protected virtual async Task<string> GetStockMessageForAttributesAsync(Product p
             if (!product.DisplayStockAvailability)
                 return string.Empty;
 
-            string stockMessage;
```

Comment:
I do not agree with such changes, it seems to me that the readability of the method will suffer greatly, instead of one exit point, we have as many as six.

Reply:
Fair enough - I personally don't agree, since the reader will have to scroll down to the end of the method and try to follow the rest of it to see that nothing more was done after all (doing exactly that was what made me change it), but that's an age-old debate and purely a matter of personal preference ;) Early returns seemed to be preferred elsewhere in the code, too. I'll change it back :)

Reply:
Thank you very much for understanding. I know that this is a discussion topic and glad that we did not go to disputes :)

Reply:
It's your code ;)